File: signature/apache.go
package signature

var ApacheSignature = Signature{
	Name:        "Apache HTTP Server",
	Description: "Apache is a free and open-source cross-platform web server software.",
	Cpe:         "cpe:/a:apache:http_server",
	DetectPattern: DetectPattern{
		Headers: []Header{{
			Name:  "server",
			Value: "Apache",
		}},
	},
	VersionPattern: VersionPattern{
		Headers: []Header{{
			Name:  "server",
			Value: `Apache/(\d+\.\d+\.\d+)`,
		}},
	},
	OnlySameHost: true,
}

File: src/signatures/technologies/apache.ts
import type { Signature } from "../_types.js";

export const apacheHttpServerSignature: Signature = {
  name: "Apache HTTP Server",
  description:
    "Apache is a free and open-source cross-platform web server software.",
  cpe: "cpe:/a:apache:http_server",
  rule: {
    confidence: "high",
    headers: {
      server: "apache/?(\\d+\\.\\d+\\.\\d+)?",
    },
  },
};
